src/NuGet.Core/NuGet.Packaging/Rules/DependeciesGroupsForEachTFMRule.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    internal class DependeciesGroupsForEachTFMRule : IPackageRule
+    {
+        private HashSet<NuGetFramework> _compatNotExactMatches;
+        private HashSet<NuGetFramework> _noExactMatchesFromFile;
+        private HashSet<NuGetFramework> _noExactMatchesFromNuspec;
+
+        public string MessageFormat { get; }
+
+        public string CompatMatchFoundWarningMessageFormat { get; }
+
+        public DependeciesGroupsForEachTFMRule(string messageFormat, string compatMatchFoundWarningMessageFormat)
```

Comment:
I know this is the pattern copied from other validations, but I really don't see the point in passing the message in, rather than just using the resources class directly here. It would save pointless duplication in the unit tests as well.